Use Node 16/18 and NPM

cmoesel opened the pull request:
This PR removes yarn in favor of npm. The `yarn.lock` is replaced by `package-lock.json`. The GitHub actions and README have also been updated. In addition, a new `test:plus` script has been added for running tests, lint, and prettier in one shot.

To test, pull the branch and delete `node_modules`. Make sure you're running Node 18, then run `npm install` followed by `npm test:plus`. Everything should work well. Note that Node 14 and 16 should also work (and are tested in CI), but we avoid them in development since they use a different package-lock format.

cmoesel committed:
Use Node 16/18 and NPM

cmoesel committed:
Fix test and lint commands for Windows

cmoesel committed:
Fix action name to reflect node version

cmoesel committed:
Change script name from 'check' to 'test:plus' for consistency w/ other cql exec libraries

cmoesel committed:
README tweaks

jafeltra commented on .github/workflows/ci-workflow.yml:
```diff
@@ -3,59 +3,33 @@ name: CI Checks
-      - name: Lint the source code
-        run: yarn lint
+          node-version: '18'
+      - run: npm install
```
This probably doesn't matter too much, but I think it is good to use `npm ci` in actions because it uses the versions specified in the `package-lock.json` and I think it's a little faster. Since you're updating the actions now anyway, I figured I'd suggest it now.

If you agree, there's another `npm install` in the `test` action.

cmoesel replied:
It's a good idea, but apparently `npm ci` is more sensitive to the `package-lock.json` format -- because it causes the Node 14 actions to fail. Since we still want to target Node 14 (for now), I think we need to stick with `npm install`.

jafeltra replied:
Agreed!

jafeltra reviewed (commented):
This looks good to me. I left one comment that you can take or leave. Besides that, I didn't see anything else in the code.

I'm not sure how strict you tend to be in PR descriptions in this repo, but just so you know, I think you're referring to the wrong npm script in the description:

> Make sure you're running Node 18, then run `npm install` followed by `npm check`

should probably be

> Make sure you're running Node 18, then run `npm install` followed by `npm run test:plus`

cmoesel committed:
Use npm ci in GitHub actions

cmoesel committed:
Back to npm install

Using npm ci chokes w/ node 14 -- so use npm install.

Also update checkout and setup-node actions to use v3.
